### PR TITLE
feat(MailerTemplates): new interface for mailer template fields

### PR DESCRIPTION
**Pull request by chillfox**

### CODE REVIEW

File: src/placeos-driver/interface/mailer_templates.cr [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  abstract def template_fields : Array(TemplateFields)
+
+  alias TemplateFields = NamedTuple(
+    # The same as is being used for #send_template.
+    # example: {"bookings", "booked_by_notify"}
+    trigger: Tuple(String, String),
+
+    # Human readable name
+    # example: "Booking booked by notification"
+    name: String,
```
Comment (stakach) [inline src/placeos-driver/interface/mailer_templates.cr:41]: Should we also add an optional description field?

**Review by stakach — CHANGES_REQUESTED** [review]
Looks good!

File: src/placeos-driver/interface/mailer_templates.cr (hunk 1 above)
Comment (chillfox) [reply src/placeos-driver/interface/mailer_templates.cr:41]: sure, can't hurt.

**Review by stakach — APPROVED** [review]
LGTM

**Pull request merged**